docs/basic/pipeline_testing.md

```diff
@@ -0,0 +1,138 @@
+          },
+          "method": {
+            "value": "GET"
+          }
+        },
+        "dependsOn": [
+          {
+            "activity": "Set JobName",
+            "dependencyConditions": [
+              "Succeeded"
+            ]
+          }
+        ]
+      }
+  ]
+}
+```
+
+Let's write a test for validating the correct flow of activities, the correct setting of the variable and the correct evaluation of the `url` property.
+
+## Arrange
+
+Get a reference to the pipeline you want to test using the `get_pipeline_by_name` method on the `Repository` instance.
+
+```python
+pipeline = test_framework.repository.get_pipeline_by_name("batch_job")
+```
+
+## Act
+
+Start evaluating the pipeline with the requested parameters. The `evaluate_pipeline` method returns a generator that yields the activities in the order they are executed.
+
+```python
+activities = test_framework.evaluate_pipeline(pipeline, [
+    RunParameter(RunParameterType.Pipeline, "JobId", "123"),
+    RunParameter(RunParameterType.Global, "BaseUrl", "https://example.com"),
+])
+```
+
+## Assert
+
+Request the next activity from the returned generator and verify that the output of the activity matches the expected result. Repeat until all activities have been evaluated.
+
+```python
+set_variable_activity = next(activities)
+assert set_variable_activity is not None
+assert "Set JobName" == set_variable_activity.name
+assert "Job-123" == activity.type_properties["value"].value
+
+get_version_activity = next(activities)
+assert get_version_activity is not None
+assert "Get version of job" == get_version_activity.name
+assert "https://example.com/Job-123/version" == get_version_activity.type_properties["url"].value
+assert "GET" == get_version_activity.type_properties["method"]
+```
+
+### Activity output references
+
+If an activity output is being referenced (e.g. `activity('Get version of job').output.version`), make sure to set the result of the activity using the `set_result` method before requesting the next activity from the generator.
+
+```python
+get_version_activity.set_result(DependencyCondition.Succeeded, { "version", "1.0.0" })
+```
+
+The `next(activities)` method might throw an exception if the expression is invalid or if a property of an expression is not (yet) available in the state. Make sure that:
+
+1. Parameters are supplied
+2. Variables are being set correctly through the `SetVariable` activity
+3. Activity outputs being referenced are set through `activity.set_result` method
+4. Dependency conditions are met
+
+### Asserting end of pipeline execution
+
+When the generator has no more activities to return, it raises a `StopIteration` exception. This is a signal that all activities have been evaluated or that no activities meet the required dependency conditions. If you expect the pipeline to be finished, it can be asserted using the `pytest.raises` context manager:
```

Comment:
```suggestion
When the generator has no more activities to return, it raises a `StopIteration` exception. This is a signal that all activities have been evaluated or that no activities meet the required dependency conditions. If the pipeline is expected to be finished, it can be asserted using the `pytest.raises` context manager:
```